thread-manager/src/policy.rs

```diff
@@ -0,0 +1,114 @@
+#[cfg(target_os = "linux")]
+use thread_priority::{NormalThreadSchedulePolicy, ThreadExt, ThreadSchedulePolicy};
+use {
+    serde::{Deserialize, Serialize},
+    std::sync::OnceLock,
+};
+
+#[cfg(not(target_os = "linux"))]
+#[derive(Clone, Copy)]
+pub(crate) struct ThreadSchedulePolicy {}
+
+static CORE_COUNT: OnceLock<usize> = OnceLock::new();
+
+pub const DEFAULT_PRIORITY: u8 = 0;
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub enum CoreAllocation {
+    ///Use OS default allocation (i.e. do not alter core affinity)
+    #[default]
+    OsDefault,
+    ///Pin each thread to a core in given range. Number of cores should be >= number of threads
+    PinnedCores { min: usize, max: usize },
+    ///Pin the threads to a set of cores
+    DedicatedCoreSet { min: usize, max: usize },
+}
+
+impl CoreAllocation {
+    /// Converts into a vector of core IDs. OsDefault is converted to empty vector.
+    pub fn as_core_mask_vector(&self) -> Vec<usize> {
+        let core_count = CORE_COUNT.get_or_init(num_cpus::get);
+        match *self {
+            CoreAllocation::PinnedCores { min, max } => (min..max).collect(),
+            CoreAllocation::DedicatedCoreSet { min, max } => (min..max).collect(),
+            CoreAllocation::OsDefault => Vec::from_iter(0..*core_count),
```

Comment:
The doc-comment above says `OsDefault` is converted to empty vector, but this is not empty (unless core_count=0).

We also only use `core_count` in this branch of the match, so would be better (imo) for that variable to have its scope limited to this branch.

Reply:
Done. Also fixed comment.